### PR TITLE
Feature/input subselect

### PR DESCRIPTION
**Pull request by nlouwen**
Contains:

- sorted GCF cutoffs to make sure singleton culling does not affect higher cutoffs
- connected components only include nodes present in the input set
- split family generation for mix/classify methods
- unique family ids in new "family" table in database

### CODE REVIEW

File: big_scape/main.py [hunk 1]
```diff
@@ -408,13 +454,14 @@ def align_callback(tasks_done: int):
...
-        )
-        query_bin.add_records(
-            [record for record in all_bgc_records if record is not None]
-        )
+        # TODO: select largest cc once, instead of rebuilding bins?
         for cutoff in run["gcf_cutoffs"]:
+            query_bin = bs_comparison.ConnectedComponentPairGenerator(
+                cc_cutoff[cutoff], "Query"
+            )
+            query_bin.add_records(
```
Comment (CatarinaCarolina) [inline big_scape/main.py:462]: could add only the records from the query_connected_component?

File: big_scape/network/families.py [hunk 2]
```diff
@@ -94,11 +98,43 @@ def save_to_db(regions_families):
...
-    DB.execute(insert_statement)
+    for region_id, family, cutoff, bin_label in regions_families:
+        # obtain unique family id if present
+        fam_id_query = (
+            select(family_table.c.id)
+            .where(family_table.c.center_id == family)
+            .where(family_table.c.cutoff == cutoff)
+            .where(family_table.c.bin_label == bin_label)
+        )
+        family_id = DB.execute(fam_id_query).fetchone()
```
Comment (CatarinaCarolina) [inline big_scape/network/families.py:112]: would this ever happen?

**Review by CatarinaCarolina — APPROVED** [review]
lgtm  (albeit superficially)

File: big_scape/main.py (hunk 1 above)
Comment (nlouwen) [reply big_scape/main.py:462]: within add_records() only records in the cc are added

**Pull request merged**